### PR TITLE
openzwave: use official version

### LINKED ISSUE #86125: open-zwave: built version is too specific
Hi, i need to use open-zwave as a dependency for a build, but the version built in the Nix package is a specific fork for supporting home-assistant, which does not work for my package build due to linking errors.

https://github.com/NixOS/nixpkgs/blob/98dbff9a65c4d0f67834787398021a636896f204/pkgs/development/libraries/openzwave/default.nix#L12

I will happily contribute a new package, but I'm not sure how to proceed, with regards to how it would fit in with this package, or if indeed the best way would be to modify this package to use the upstream repo.

Pinging people who have maintained the package, apologies for the noise!

@etu 
@volth 
@dotlambda 
@OmnipotentEntity 
@matthewbauer 
@WhittlesJr

### PR DESCRIPTION
**Pull request by dotlambda**
###### Motivation for this change
fixes https://github.com/NixOS/nixpkgs/issues/86125

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

We should unvendor openzwave's dependencies as in https://github.com/archlinux/svntogit-community/blob/packages/openzwave/trunk/openzwave-system-libs.patch.

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 118391` at fa2926ef run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages built successfully:</summary>
  <ul>
    <li>home-assistant</li>
    <li>openzwave</li>
    <li>python38Packages.homeassistant-pyozw</li>
    <li>python38Packages.python_openzwave</li>
    <li>python39Packages.homeassistant-pyozw</li>
    <li>python39Packages.python_openzwave</li>
  </ul>
</details>

**Review by WhittlesJr — APPROVED** [review]

File: pkgs/development/python-modules/homeassistant-pyozw/default.nix [hunk 1]
```diff
@@ -12,4 +12,17 @@ python_openzwave.overridePythonAttrs (oldAttrs: rec {
 
   patches = [];
   meta.homepage = "https://github.com/home-assistant/python-openzwave";
-})
+})).override {
+  openzwave = openzwave.overrideAttrs (oldAttrs: {
+    version = "2020-03-24";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/homeassistant-pyozw/default.nix:17]:
```suggestion
    version = "unstable-2020-03-24";
```

**Review by SuperSandro2000 — APPROVED** [review]
other than that LGTM

**Comment by SuperSandro2000** [comment]
Also can we put the doc into a separate output? It is quite massive.

```
openzwave: Ask if output path share/doc (117.2M) could be split with outputs = [ ... "doc" ];
```

**Review by nuxeh — APPROVED** [review]

**Comment by dotlambda** [comment]
I noticed that `home-assistant` now depends on both versions of `openzwave`. Any idea why?

**Comment by dotlambda** [comment]
> I noticed that `home-assistant` now depends on both versions of `openzwave`. Any idea why?

Turns out this is solved by switching the order of `override` and `overridePythonAttrs`.

**Comment by dotlambda** [comment]
> Also can we put the doc into a separate output?

Done.

**Pull request merged**